var/spack/repos/builtin/packages/intel-oneapi-compilers/package.py

```diff
@@ -113,6 +114,15 @@ class IntelOneapiCompilers(IntelOneApiPackage):
 
     depends_on("patchelf", type="build")
 
+    # TODO: effectively gcc is a direct dependency of intel-oneapi-compilers, but we
+    # cannot express that properly. For now, add conflicts for non-gcc compilers
+    # instead.
+    for __compiler in spack.compilers.supported_compilers():
+        if __compiler != "gcc":
```

Comment:
I see where you're coming from, this may be the best we can do for now, but is it that gcc is a direct dependency or that we need the system toolchain that's in use? The latter might let us propagate it transitively through other compilers as long as we can query them for the appropriate thing.

Reply:
More concretely, for clang-based compilers for example we could use the `Selected GCC installation` from clang -v, or the result of `dirname $(clang -print-libgcc-file-name)`.

Reply:
It's also a direct dependency, althought that's not covered by this PR:

```
$ readelf -d ./compiler/2022.2.0/linux/lib/libsycl.so | grep 'libgcc_s.so'
 0x0000000000000001 (NEEDED)             Shared library: [libgcc_s.so.1]
```

Reply:
So, I was thinking to submit another PR that would `patchelf --add-rpath <gcc compiler support library dir> <intel oneapi compiler support library>`

Reply:
That makes sense, but all the places this is happening only add the GCC stuff if the compiler is gcc, so compiling with clang we still don't get a toolchain set right?

Reply:
Yeah, here we just add conflicts to only ever allow GCC as a dep

Reply:
Regarding `llvm%clang`, we could introspect `clang` and have it reports its chosen gcc toolchain, and use that to fix `GCC_INSTALL_PREFIX` once and for all too... but that would another PR.